lib/segment/src/types.rs

```diff
@@ -1221,23 +1227,42 @@ impl GeoRadius {
 
 /// Geo filter request
 ///
-/// Matches coordinates inside the polygon, defined by the given coordinates in order.
+/// Matches coordinates inside the polygon, defined by exterior and interiors.
 #[derive(Debug, Deserialize, Serialize, JsonSchema, Clone, PartialEq)]
 #[serde(rename_all = "snake_case")]
 pub struct GeoPolygon {
-    /// Ordered list of coordinates representing the vertices of a polygon.
-    pub points: Vec<GeoPoint>,
+    /// An ordered list of lists of GeoPoint coordinates, arranged in order.
+    /// For Polygons with more than one of these rings, the first MUST be
+    /// the exterior ring, and any others MUST be interior rings. The exterior
+    /// ring bounds the surface, and the interior rings (if present) bound holes
+    /// within the surface.
+    pub rings: Vec<GeoLineString>,
```

Comment:
Why not create two fields?

Reply:
The [GeoJSON spec](https://datatracker.ietf.org/doc/html/rfc7946#section-3.1.6) defines as first ring and other rings. So I think have gRPC proto same as GeoJSON can be more cleaner.

Reply:
Oh, didn't know about this, might be worth to explicitly say this is following GeoJSON spec.

Reply:
Yes this makes sense! added.